code/modules/cooking_with_jane/cooking.dm

```diff
@@ -0,0 +1,793 @@
+
+						if(J_MED)
+							if(step_list[3] > CWJ_BURN_TIME_MEDIUM)
+								reason="Time too large for Medium setting on CWJ_USE_GRILL; Food will automatically burn."
+
+						if(J_HI)
+							if(step_list[3] > CWJ_BURN_TIME_HIGH)
+								reason="Time too large for High setting on CWJ_USE_GRILL; Food will automatically burn."
+
+						else
+							reason="Unrecognized temperature for CWJ_USE_GRILL"
+
+					if(!reason)
+						create_step_use_grill(step_list[2], step_list[3], FALSE)
+
+				if(CWJ_USE_GRILL_OPTIONAL)
+					if(step_list.len < 3)
+						reason="Bad argument Length for CWJ_USE_GRILL_OPTIONAL"
+					switch(step_list[2])
+						if(J_LO)
+							if(step_list[3] > CWJ_BURN_TIME_LOW)
+								reason="Time too large for Low setting on CWJ_USE_GRILL_OPTIONAL; Food will automatically burn."
+
+						if(J_MED)
+							if(step_list[3] > CWJ_BURN_TIME_MEDIUM)
+								reason="Time too large for Medium setting on CWJ_USE_GRILL_OPTIONAL; Food will automatically burn."
+
+						if(J_HI)
+							if(step_list[3] > CWJ_BURN_TIME_HIGH)
+								reason="Time too large for High setting on CWJ_USE_GRILL_OPTIONAL; Food will automatically burn."
+						else
+							reason="Unrecognized temperature for CWJ_USE_GRILL_OPTIONAL"
+					if(!reason)
+						create_step_use_grill(step_list[2], step_list[3], TRUE)
+				if(CWJ_USE_OVEN)
+					if(step_list.len < 3)
+						reason="Bad argument Length for CWJ_USE_OVEN"
+					switch(step_list[2])
+						if(J_LO)
+							if(step_list[3] > CWJ_BURN_TIME_LOW)
+								reason="Time too large for Low setting on CWJ_USE_OVEN; Food will automatically burn."
+
+						if(J_MED)
+							if(step_list[3] > CWJ_BURN_TIME_MEDIUM)
+								reason="Time too large for Medium setting on CWJ_USE_OVEN; Food will automatically burn."
+
+						if(J_HI)
+							if(step_list[3] > CWJ_BURN_TIME_HIGH)
+								reason="Time too large for High setting on CWJ_USE_OVEN; Food will automatically burn."
+
+						else
+							reason="Unrecognized temperature for CWJ_USE_OVEN"
+
+					if(!reason)
+						create_step_use_oven(step_list[2], step_list[3], FALSE)
+				if(CWJ_USE_OVEN_OPTIONAL)
+					if(step_list.len < 3)
+						reason="Bad argument Length for CWJ_USE_OVEN_OPTIONAL"
+					switch(step_list[2])
+						if(J_LO)
+							if(step_list[3] > CWJ_BURN_TIME_LOW)
+								reason="Time too large for Low setting on CWJ_USE_OVEN_OPTIONAL; Food will automatically burn."
+
+						if(J_MED)
+							if(step_list[3] > CWJ_BURN_TIME_MEDIUM)
+								reason="Time too large for Medium setting on CWJ_USE_OVEN_OPTIONAL; Food will automatically burn."
+
+						if(J_HI)
+							if(step_list[3] > CWJ_BURN_TIME_HIGH)
+								reason="Time too large for High setting on CWJ_USE_OVEN_OPTIONAL; Food will automatically burn."
+						else
+							reason="Unrecognized temperature for CWJ_USE_OVEN_OPTIONAL"
+					if(!reason)
+						create_step_use_oven(step_list[2], step_list[3], TRUE)
+
+			//Named Arguments modify the recipe in fixed ways
```

Comment:
If `reason` is set before this line, it might be erased in the following because they have `reason =`

Reply:
Gimme a bit and I'll give it a local test.

Reply:
I think it won't cause a problem, it's just that if you have two issues, you will miss the first one because it will be overwritten by the reason of the second one.

Reply:
Y'know, I'll look into this. There is a rare issue with the tracker where sometimes, it doesn't... register a step as completed. I anecdotally thought it was related to reagents, but I'll look further into this to see if this could be a culprit.